lib/vsc/administration/slurm/sync.py

```diff
@@ -342,6 +573,67 @@ def slurm_vo_accounts(account_page_vos, slurm_account_info, clusters, host_insti
     return commands
 
 
+def slurm_project_users_accounts(project_members, active_accounts, slurm_user_info, clusters):
+    """Check if the users are in the project account.
+
+    For users in the project:
+
+    - If the user does not exist on the system:
+        - a new association in the default account is created. The associated QoS does not allow jobs.
+        - a new association in the project account is created.
+
+    - For users who left the project:
+        - The user's association in the project is removed.
+        - If this was the last project for this user, we also remove the association in the default account.  (TODO)
+    """
+
+    commands = []
+
+    for cluster in clusters:
+        cluster_users_acct = [
+            (user.User, user.Account) for user in slurm_user_info if user and user.Cluster == cluster
+        ]
+
+        new_users = set()
+        remove_project_users = set()
+        all_project_users = set()
+
+        for (members, project_name) in project_members:
+
+            # these are the current Slurm users for this project
+            slurm_project_users = set([user for (user, acct) in cluster_users_acct if acct == project_name])
+            all_project_users |= slurm_project_users
+
+            # these users are not yet in the Slurm DBD for any project
+            new_users |= set([(user, project_name) for user in (members & active_accounts) - slurm_project_users])
+
+            # these are the Slurm users that should no longer be associated with the project
+            remove_project_users |= set([(user, project_name) for user in slurm_project_users - members])
+
+        logging.info("%d new users", len(new_users))
+        logging.info("%d removed project users", len(remove_project_users))
+
+        # these are the users not in any project, we should decide if we want any of those
+        remove_slurm_users = set([u[0] for u in cluster_users_acct]) - all_project_users
+
+        if remove_slurm_users:
+            logging.warning(
+                "Number of slurm users not in projecs: %d > 0: %s", len(remove_slurm_users), remove_slurm_users
```

Comment:
or 
```
logging.error("%s user not in any project: %s", len(remove_slurm_users), remove_slurm_users)
```

Reply:
and `raise` ofcourse

Reply:
No, not yet. I'll fix this in a follow up PR, after I think on how to deal with gadminforever.

Reply:
give it a project ilek everyone else or pass a list of poeple with no project